cmd/controller/main.go

```diff
@@ -101,95 +163,60 @@ func main() { // nolint: gocyclo
 		WithField("alwaysPullSidecarImage", alwaysPullSidecar).
 		WithField("Version", pkg.Version).Info("starting gameServer operator...")
 
-	if minPort <= 0 || maxPort <= 0 {
-		logger.Fatal("Min Port and Max Port values are required.")
-	} else if maxPort < minPort {
-		logger.Fatal("Max Port cannot be set less that the Min Port")
-	}
-
-	config, err := rest.InClusterConfig()
-	if err != nil {
-		logger.WithError(err).Fatal("Could not create in cluster config")
+	return ctlConfig{
+		minPort:           minPort,
+		maxPort:           maxPort,
+		sidecarImage:      sidecarImage,
+		alwaysPullSidecar: alwaysPullSidecar,
+		keyFile:           keyFile,
+		certFile:          certFile,
 	}
+}
 
-	kubeClient, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		logger.WithError(err).Fatal("Could not create the kubernetes clientset")
-	}
+// ctlConfig stores all required configuration to create a game server controller.
+type ctlConfig struct {
```

Comment:
Super minor nit - should this just be `config` - it's not like we have any other configurations around? Happy if you ignore this, but thought I'd ask.

Reply:
Sure. I'll rename it.